grid.v3/server.go

```diff
@@ -106,6 +102,10 @@ func (s *Server) Serve(lis net.Listener) error {
 	s.ctx = ctx
 	s.cancel = cancel
 
+	// Start the registry and monitor that it is
+	// running correctly.
+	registryErrors := s.monitorRegistry(lis.Addr())
```

Comment:
Function `monitorRegistry` uses the `s.ctx` so it needs to be run after it's set.